mne/epochs.py

```diff
@@ -3249,6 +3260,42 @@ def __init__(
         # keep track of original sfreq (needed for annotations)
         raw_sfreq = raw.info["sfreq"]
 
+        # get events from annotations if no events given
+        if events is None:
+            events, event_id_tmp = events_from_annotations(raw)
+            if events.size == 0:
+                raise RuntimeError(
+                    "No usable annotations found in the raw object. "
+                    "Either `events` must be provided or the raw "
+                    "object must have annotations to construct epochs"
+                )
+            if any(raw.annotations.duration > 0):
+                logger.info(
+                    "Ignoring annotation durations, only fixed "
+                    "duration epochs are currently supported"
+                )
+            if event_id is None:
+                event_id = event_id_tmp
+            # if event_id is the names of events, map to events integers
+            elif isinstance(event_id, (str, list, tuple)):
+                if isinstance(event_id, str):
+                    event_id = [event_id]
+                if all([my_id in event_id_tmp for my_id in event_id]):
```

Comment:
[nitpick] I find this logic a bit easier to follow

```suggestion
            if isinstance(event_id, str):
                event_id = [event_id]
            if isinstance(event_id, (list, tuple)):
                if set(event_id).issubset(set(event_id_tmp)):
```